app/controllers/users_controller.rb

```diff
@@ -169,15 +169,15 @@ def remove_identity
   def onboarding_update
     authorize User
 
-    user_params = { saw_onboarding: true }
+    user_params = {}
 
     if params[:user]
-      if params.dig(:user, :username).blank?
+      if params[:user].key?(:username) && params[:user][:username].blank?
```

Comment:
Wouldn't this work instead, it's pretty much what `Hash#dig` (and in this specific case [ActionController::Parameter#dig](https://api.rubyonrails.org/classes/ActionController/Parameters.html)) was made for:

```suggestion
      if params.dig(:user, :username).blank?
```

Reply:
I don't think so - in this case I only want to return an error if the value was provided and it was empty - I don't want to validate a key that wasn't sent in the body ("username" is only sent on a single panel, so every other patch request was getting the error response though nothing was wrong).

In this case I'm changing the condition, not trying to reimplement safe hash navigation